pallets/loans/src/lib.rs

```diff
@@ -629,6 +629,31 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Sets a new liquidation incentive percentage for `currency_id`.
+        ///
+        /// Does nothing if the provided asset is not attached to an existent incentive.
+        ///
+        /// - `currency_id`: the asset that is going to be modified.
+        #[pallet::weight(T::WeightInfo::repay_borrow_all())]
+        #[transactional]
+        pub fn set_liquidation_incentive(
+            origin: OriginFor<T>,
+            currency_id: CurrencyId,
+            liquidate_incentive: Ratio,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            Self::ensure_currency(&currency_id)?;
+
+            let does_not_have_incentive = <LiquidationIncentive<T>>::try_get(currency_id).is_err();
+            if does_not_have_incentive {
+                return Ok(().into());
+            }
+
+            <LiquidationIncentive<T>>::insert(currency_id, liquidate_incentive);
+
+            Ok(().into())
```

Comment:
Can we remove `.into()`

Reply:
Why we can't use only `Ok(())`?

Reply:
`pub type DispatchResultWithPostInfo = sp_runtime::DispatchResultWithInfo<crate::weights::PostDispatchInfo>;`
`pub type DispatchResultWithInfo<T> = sp_std::result::Result<T, DispatchErrorWithPostInfo<T>>;`

`DispatchResultWithPostInfo` expects `Ok(crate::weights::PostDispatchInfo)`, thus the reason why `Ok(())` doesn't work.

`().into()` works here because of

```rust
impl From<()> for PostDispatchInfo {
	fn from(_: ()) -> Self {
		Self {
			actual_weight: None,
			pays_fee: Default::default(),
		}
	}
}
```

In any case, the call should return an actual weight instead of `PostDispatchInfo::default`.